File: leetcode/0172. Factorial Trailing Zeroes/Solution.swift
// https://leetcode.com/problems/factorial-trailing-zeroes/

class Solution {
    func trailingZeroes(_ n: Int) -> Int {
        if n == 0 { return 0 }
        return n / 5 + trailingZeroes(n / 5)
    }
}

File: leetcode/0172. Factorial Trailing Zeroes/Solution.py
# https://leetcode.com/problems/factorial-trailing-zeroes/

class Solution:
    def trailingZeroes(self, n: int) -> int:
        if n == 0:
        	return 0
        
        return n // 5 + self.trailingZeroes(n // 5)

s = Solution()
assert s.trailingZeroes(3) == 0
assert s.trailingZeroes(5) == 1